staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -37,42 +37,62 @@ import (
 	"k8s.io/klog"
 )
 
-// main demonstrates a leader elected process that will step down if interrupted.
+func buildConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig != "" {
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
 func main() {
 	klog.InitFlags(nil)
+
+	var kubeconfig string
+	var resourceName string
+	var ttlSeconds int
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&resourceName, "resource-name", "example", "The leader election resource name (Configmap or Endpoint name)")
+	flag.IntVar(&ttlSeconds, "ttl", 30, "TTL for leader election in seconds")
 	flag.Parse()
-	args := flag.Args()
-	if len(args) != 3 {
-		log.Fatalf("requires three arguments: ID NAMESPACE CONFIG_MAP_NAME (%d)", len(args))
+
+	podName := os.Getenv("POD_NAME")
+	podNamespace := os.Getenv("POD_NAMESPACE")
```

Comment:
any particular reason to use environment variable, instead of flags?

Reply:
If this example runs in Kubernetes cluster, we can pass Pod name to identify it, but I think this can replace to flags.